File: server/src/main/kotlin/wa2/polito/it/letduchidegliabruzzi/server/dal/dao/ticket/Ticket.kt
package wa2.polito.it.letduchidegliabruzzi.server.dal.dao.ticket

import jakarta.persistence.*
import wa2.polito.it.letduchidegliabruzzi.server.dal.dao.chat.Chat
import wa2.polito.it.letduchidegliabruzzi.server.dal.dao.product.Product
import wa2.polito.it.letduchidegliabruzzi.server.dal.dao.status_history.StatusHistory

@Entity
@Table(name = "ticket")
class Ticket(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val ticketID: Int?,
    val description: String,
    val status: String,
    val priority: String?,
    val createdAt: String,
    val customerUsername: String,
    val expertUsername: String?,
    @ManyToOne @JoinColumn(name = "ean") val product: Product,
    @OneToMany val statusHistory: List<StatusHistory>,
    @OneToOne @JoinColumn(name = "ticketID") val chat: Chat?
)

File: client/src/models/Ticket.js
class Ticket {
    ticketID
    description
    status
    priority
    createdAt
    productEan
    productBrand
    productName
    customerUsername
    customerName
    customerSurname
    employeeUsername
    employeeName
    employeeSurname
    statusHistory
    chatID

    constructor(ticketID, description, status, priority, createdAt, productEan, productBrand, productName, customerUsername, customerName, customerSurname, employeeUsername, employeeName, employeeSurname, statusHistory, chatID) {
        this.ticketID = ticketID;
        this.description = description;
        this.status = status;
        this.priority = priority;
        this.createdAt = createdAt;
        this.productEan = productEan;
        this.productBrand = productBrand;
        this.productName = productName;
        this.employeeUsername = employeeUsername;
        this.employeeName = employeeName;
        this.employeeSurname = employeeSurname;
        this.customerUsername = customerUsername;
        this.customerName = customerName;
        this.customerSurname = customerSurname;
        this.statusHistory = statusHistory;
        this.chatID = chatID;
    }
}

export default Ticket